main.go

```diff
@@ -57,7 +57,7 @@ func main() {
 		MetricsBindAddress: metricsAddr,
 		Port:               9443,
 		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "1fc40399.io",
+		LeaderElectionID:   "external-secrets-controller",
```

Comment:
Well this needs to be a known value so the deployment manifests/charts grant the RBAC permissions needed to create/patch this resource, otherwise the controller can't acquire the lock. The name can be configurable via flag if needed, but this can cause some edge cases where the controller has two leaders when the lock name has changed between deployment versions.

Controller-runtime v0.7.0 has the option to use the newer lease resource for the controller lock rather than configmap, when upgrading to that this should help fix conflicting names here.

Reply:
Thanks for the explanation :)